hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/repair/om/OMRatisLogRepair.java

```diff
@@ -0,0 +1,300 @@
+          OzoneManagerProtocolProtos.OMRequest oldRequest = OMRatisHelper
+              .convertByteStringToOMRequest(next.getStateMachineLogEntry().getLogData());
+          OzoneManagerProtocolProtos.OMRequest.Builder newRequest = OzoneManagerProtocolProtos.OMRequest.newBuilder()
+              .setCmdType(EchoRPC)
+              .setClientId(oldRequest.getClientId())
+              .setEchoRPCRequest(OzoneManagerProtocolProtos.EchoRPCRequest.newBuilder().build());
+
+          if (oldRequest.hasUserInfo()) {
+            newRequest.setUserInfo(oldRequest.getUserInfo());
+          }
+          if (oldRequest.hasTraceID()) {
+            newRequest.setTraceID(oldRequest.getTraceID());
+          }
+          if (oldRequest.hasLayoutVersion()) {
+            newRequest.setLayoutVersion(oldRequest.getLayoutVersion());
+          }
+          if (oldRequest.hasVersion()) {
+            newRequest.setVersion(oldRequest.getVersion());
+          }
+
+          RaftProtos.StateMachineLogEntryProto oldEntry = next.getStateMachineLogEntry();
+          RaftProtos.StateMachineLogEntryProto.Builder newEntry =
+              RaftProtos.StateMachineLogEntryProto.newBuilder()
+                  .setCallId(oldEntry.getCallId())
+                  .setClientId(oldEntry.getClientId())
+                  .setType(oldEntry.getType())
+                  .setLogData(OMRatisHelper.convertRequestToByteString(newRequest.build()));
+          if (oldEntry.hasStateMachineEntry()) {
+            newEntry.setStateMachineEntry(oldEntry.getStateMachineEntry());
+          }
+
+          RaftProtos.LogEntryProto newLogEntry = RaftProtos.LogEntryProto.newBuilder()
+              .setTerm(next.getTerm())
+              .setIndex(next.getIndex())
+              .setStateMachineLogEntry(newEntry)
+              .build();
+
+          if (!isDryRun()) {
+            outputStream.write(newLogEntry);
+            outputStream.flush();
+          }
+          info("Replaced {" + oldRequest + "} with EchoRPC command at index "
+              + next.getIndex());
+        }
+      }
+
+      if (!isDryRun()) {
+        outputStream.flush();
+        outputStream.close();
+        Files.move(outputFile.toPath(), segmentFile.toPath(), StandardCopyOption.REPLACE_EXISTING);
+        info("Moved temporary output file to correct raft log location : " + segmentFile.toPath());
+      }
+
+    } catch (Exception ex) {
+      error("Exception: " + ex);
+    } finally {
+      if (logInputStream != null) {
+        logInputStream.close();
+      }
+      if (outputStream != null) {
+        outputStream.flush();
+        outputStream.close();
+      }
+      if (isDryRun()) {
+        boolean success = outputFile.delete();
+        if (!success) {
+          error("Error: Could not delete temporary output file \"" + outputFile + "\".");
+        }
+      }
+    }
+  }
+
+  private File createOutputFile(String name) throws IOException {
```

Comment:
I think we can replace this method with `Files#createTempFile`. We can do best effort cleanup on exit, otherwise the file would need to be manually deleted.